scripts/clean_labelled_passages_directory.py

```
@@ -0,0 +1,105 @@
+# This is a script for clearing down the labelled passages directory of outputs that
+# were from running inference on non-English language documents.
+#
+# The script will:
+# 1. Find all the non-english language documents from the embeddings input prefix.
+# 2. Delete the non-english language documents from the labelled passages prefix;
+#   using the corresponding file stems.
+#
+# E.g. embeddings_input/CCLW.exec.1.1["languages"] != ["en"]
+#   -> delete labelled_passages/Q123/v1/CCLW.exec.1.1.json
+
+import json
+from pathlib import Path
+from typing import Any, Dict, Generator
+
+import boto3
+import typer
+
+
+def list_s3_objects(bucket_name: str, prefix: str) -> Generator[str, None, None]:
+    """List all objects in an S3 bucket."""
+    typer.echo(f"Listing objects in {bucket_name} with prefix {prefix}")
+    s3_client = boto3.client("s3")
+    paginator = s3_client.get_paginator("list_objects_v2")
+
+    for page in paginator.paginate(Bucket=bucket_name, Prefix=prefix):
+        for obj in page.get("Contents", []):
+            yield obj["Key"]
+
+
+def load_json_from_s3(bucket_name: str, key: str) -> Dict[str, Any]:
+    """Load JSON content from an S3 object."""
+    s3_client = boto3.client("s3")
+    response = s3_client.get_object(Bucket=bucket_name, Key=key)
+    content = response["Body"].read().decode("utf-8")
+    return json.loads(content)
+
+
+def json_generator(
+    bucket_name: str, prefix: str
+) -> Generator[tuple[Dict[str, Any], str], None, None]:
+    """Generator that yields JSON objects from S3 bucket."""
+    for key in list_s3_objects(bucket_name, prefix):
+        try:
+            yield load_json_from_s3(bucket_name, key), key
+        except json.JSONDecodeError:
+            print(f"Skipping {key}: Invalid JSON format")
+        except Exception as e:
+            print(f"Skipping {key}: {e}")
```

Are we expecting these errors as normal behaviour? I'm just wondering, if we're catching all of them, are we not running the risk of leaving non-English documents in the prefix unnoticed. Wouldn't it be better to throw these errors?

---

Agreed, makes total sense!